app/webpacker/controllers/order_cycle_controller.js

```diff
@@ -0,0 +1,52 @@
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  static targets = ['input', 'modalConfirm'];
+  static values = { initVals: { type: Object, default: {} }, hasSchedule: { type: Boolean, default: false } };
+
+  connect() {
+    if(!this.hasScheduleValue) return;
+    // Attach update callback method
+    window.adminOrderCycleUpdateCallback = this.updateCallback.bind(this);
+  }
+
+  toggleSaveBtns() {
+    if(!this.hasScheduleValue) return;
+
+    // Check that datetime input value has a change
+    const dirty = this.inputTargets.some(ele =>
+      new Date(this.initValsValue[`${ele.name}`]).getTime() !== new Date(ele.value).getTime());
```

Comment:
Instead of having to pass the current date in as a variable, we can actually ask the DOM by using `ele.defaultValue`. See example here:

https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/webpacker/controllers/bulk_form_controller.js#L163


Then you won't need to add the `initVals` values and the HTML can be cleaner.